Index: nginx-1.18.0/debian/control
===================================================================
--- nginx-1.18.0.orig/debian/control
+++ nginx-1.18.0/debian/control
@@ -1,10 +1,7 @@
 Source: nginx
 Section: httpd
 Priority: optional
-Maintainer: Ubuntu Developers <ubuntu-devel-discuss@lists.ubuntu.com>
-XSBC-Original-Maintainer: Debian Nginx Maintainers <pkg-nginx-maintainers@alioth-lists.debian.net>
-Uploaders: Christos Trochalakis <ctrochalakis@debian.org>,
-           Ondřej Nový <onovy@debian.org>,
+Maintainer: nidor1998 <nidor1998@gmail.com>
 Build-Depends: debhelper-compat (= 13),
                dpkg-dev (>= 1.15.5),
                libexpat-dev,
@@ -13,6 +10,7 @@ Build-Depends: debhelper-compat (= 13),
                libhiredis-dev,
                libmaxminddb-dev,
                libmhash-dev,
+               libmodsecurity-dev,
                libpam0g-dev,
                libpcre3-dev,
                libperl-dev,
@@ -22,9 +20,7 @@ Build-Depends: debhelper-compat (= 13),
                quilt,
                zlib1g-dev
 Standards-Version: 4.5.0
-Homepage: https://nginx.net
-Vcs-Git: https://salsa.debian.org/nginx-team/nginx.git
-Vcs-Browser: https://salsa.debian.org/nginx-team/nginx
+Homepage: https://github.com/nidor1998/libnginx-mod-http-modsecurity
 Rules-Requires-Root: no
 
 Package: nginx
@@ -482,3 +478,15 @@ Description: RTMP support for Nginx
   - Stream relay support via a push or pull model;
   - Integrated stream recording;
   - and more.
+
+Package: libnginx-mod-http-modsecurity
+Architecture: any
+Depends: libmodsecurity3,
+         ${misc:Depends},
+         ${shlibs:Depends}
+Recommends: nginx,modsecurity-crs
+Description: WAF module for Nginx
+ The ModSecurity-nginx connector is the connection point between nginx and
+ libmodsecurity (ModSecurity v3). Said another way, this project provides a
+ communication channel between nginx and libmodsecurity. This connector is
+ required to use LibModSecurity with nginx.
Index: nginx-1.18.0/debian/dh_nginx
===================================================================
--- nginx-1.18.0.orig/debian/dh_nginx
+++ nginx-1.18.0/debian/dh_nginx
@@ -33,7 +33,7 @@ dh_nginx - register configuration snippe
 
 sub nginx_depends
 {
-	return 'nginx-common (= ${source:Version})'
+	return 'nginx-common (>= ${source:Upstream-Version}), nginx-common (<< ${source:Upstream-Version}.1~)'
 }
 
 sub nginx_api_installdir
Index: nginx-1.18.0/debian/modules/control
===================================================================
--- nginx-1.18.0.orig/debian/modules/control
+++ nginx-1.18.0/debian/modules/control
@@ -64,3 +64,6 @@ Homepage: https://github.com/arut/nginx-
 Files-Excluded: test
 Version: 1.2.1
 
+Module: http-modsecurity
+Homepage: https://github.com/owasp-modsecurity/ModSecurity-nginx
+Version: 1.0.4
Index: nginx-1.18.0/debian/rules
===================================================================
--- nginx-1.18.0.orig/debian/rules
+++ nginx-1.18.0/debian/rules
@@ -16,6 +16,7 @@ DYN_MODS := \
 	http-geoip2 \
 	http-headers-more-filter \
 	http-image-filter \
+	http-modsecurity \
 	http-ndk \
 	http-perl \
 	http-subs-filter \
@@ -118,6 +119,7 @@ extras_configure_flags := \
 			--add-dynamic-module=$(MODULESDIR)/http-auth-pam \
 			--add-dynamic-module=$(MODULESDIR)/http-cache-purge \
 			--add-dynamic-module=$(MODULESDIR)/http-dav-ext \
+			--add-dynamic-module=$(MODULESDIR)/http-modsecurity \
 			--add-dynamic-module=$(MODULESDIR)/http-ndk \
 			--add-dynamic-module=$(MODULESDIR)/http-echo \
 			--add-dynamic-module=$(MODULESDIR)/http-fancyindex \
Index: nginx-1.18.0/debian/copyright
===================================================================
--- nginx-1.18.0.orig/debian/copyright
+++ nginx-1.18.0/debian/copyright
@@ -135,6 +135,12 @@ Files: debian/modules/rtmp/*
 Copyright: 2012-2014, Roman Arutyunyan
 License: BSD-2-clause
 
+Files: debian/modules/http-modsecurity/*
+License: Apache
+
+License: Apache
+ /usr/share/common-licenses
+
 License: BSD-2-clause
  All rights reserved.
  .
Index: nginx-1.18.0/debian/changelog
===================================================================
--- nginx-1.18.0.orig/debian/changelog
+++ nginx-1.18.0/debian/changelog
@@ -1,3 +1,9 @@
+nginx (1.18.0-6ubuntu14.7+connector104) jammy; urgency=medium
+
+  * Unofficial build for ModSecurity-nginx connector v1.0.4
+
+ -- nidor1998 <nidor1998@gmail.com>  Sun, 21 Dec 2025 06:53:16 +0000
+
 nginx (1.18.0-6ubuntu14.7) jammy-security; urgency=medium
 
   * SECURITY UPDATE: data leak in ngx_mail_smtp_module
